src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategy.kt

```diff
@@ -852,16 +855,17 @@ private class MonitorTracker(nThreads: Int) {
     fun releaseMonitor(monitor: Any) {
         // Decrement the reentrancy depth and remove the acquisition info
         // if the monitor becomes free to acquire by another thread.
-        val ai = acquiredMonitors[monitor]!!
-        ai.timesAcquired--
-        if (ai.timesAcquired == 0) acquiredMonitors.remove(monitor)
+        val info = acquiredMonitors[monitor]!!
+        info.timesAcquired--
+        if (info.timesAcquired == 0)
+            acquiredMonitors.remove(monitor)
     }
 
     /**
      * Returns `true` if the corresponding threads is waiting on some monitor.
      */
     fun isWaiting(iThread: Int): Boolean {
-        val monitor = acquiringMonitors[iThread] ?: return false
```

Comment:
Why was it "acquiringMonitors"? Why did the code work at all?

Reply:
Here I just renamed `acquiringMonitors` into `waitingMonitors`.

The `MonitorTracker`  class maintains two mappings (previously names as):
- `acquiredMonitors` mapping each locked object to `MonitorAcquiringInfo`
- `acquiringMonitors` mapping each thread to object it wants to lock (but cannot because it acquired by some another thread).

These two mappings had very similar names, that were easy to confuse. Thus I renamed `acquiringMonitors` into `waitingMonitors`. Other naming suggestions are welcome.

Reply:
Thanks for the clarification. Can we have more than one monitor in `waitingMonitors`?

Reply:
No according to the current contract, because while thread is waiting to acquire some lock, it is blocked and cannot make further progress (and thus make an attempt to acquire another lock).

Reply:
Let's rename to `waitingMonitor` for clarity

Reply:
Done